packages/tailwind/src/tailwind.tsx

```diff
@@ -26,156 +27,93 @@ export interface TailwindProps {
   config?: TailwindConfig;
 }
 
-interface EmailElementProps {
+export interface EmailElementProps {
   children?: React.ReactNode;
   className?: string;
   style?: React.CSSProperties;
 }
 
-const isComponent = (
-  element: React.ReactElement,
-): element is React.ReactElement<unknown, React.FC<unknown>> => {
-  return (
-    typeof element.type === "function" ||
-    // @ts-expect-error - we know this is a component that may have a render function
-    element.type.render !== undefined
-  );
-};
-
 export const Tailwind: React.FC<TailwindProps> = ({ children, config }) => {
-  const { stylePerClassMap, nonInlinableClasses, sanitizedMediaQueries } =
-    useTailwindStyles(children, config ?? {});
-
-  const inline = useStyleInlining(stylePerClassMap);
-
-  const nonInlineStylesToApply = sanitizedMediaQueries.filter(
-    (style) => style.trim().length > 0,
-  );
-
-  const hasNonInlineStylesToApply = nonInlineStylesToApply.length > 0;
-  let hasAppliedNonInlineStyles = false as boolean;
-
-  function processElement(
-    element: React.ReactElement<EmailElementProps>,
-  ): React.ReactElement<EmailElementProps> {
-    const propsToOverwrite = {} as Partial<EmailElementProps>;
-
-    if (!hasAppliedNonInlineStyles && hasNonInlineStylesToApply) {
-      if (element.type === "head") {
-        hasAppliedNonInlineStyles = true;
-
-        /*                   only minify here since it is the only place that is going to be in the DOM */
-        const styleElement = (
-          <style>{minifyCss(nonInlineStylesToApply.join(""))}</style>
-        );
-
-        return React.cloneElement(
-          element,
-          element.props,
-          element.props.children,
-          styleElement,
-        );
+  const tailwind = useTailwind(config ?? {});
+
+  const cloneElementWithInlinedStyles =
+    useCloneElementWithInlinedStyles(tailwind);
+
+  const nonInlineStylesRootToApply = new Root();
+  let mediaQueryClassesForAllElement: string[] = [];
+  let hasNonInlineStylesToApply = false as boolean;
```

Comment:
Do we need to cast?

Reply:
The problem here is linting. If we use a proper type for the variable eslint complains about types that can be inferred, and if we don't cast, eslint complains that the condition down below is unnecessary, when it actually is necessary.

https://github.com/resend/react-email/blob/560c789acca66d1a7bc68529443efdc69a87cd98/packages/tailwind/src/tailwind.tsx#L66-L68

The best option I found to not disable linting here was to cast the type.